### DISCUSSION TITLE
Session's management

### DISCUSSION THREAD

**Discussion by geomagilles** [kwebio/kweb-core#495] [General]
Hi, 

If I understand correctly, currently all users share the same state on the server. Can you please provide an example of how to manage user-specific states (for each user to have its own state) - ideally using ktor session management? 

Thanks,

Gilles

**Comment by sanity** [comment]
Hi, issues of state and lifecycle are left up to the user of the Kweb library right now for maximum flexibility - although it would be nice to provide a recommended approach. 

The scope of KVars and KVals are defined by where they are declared, if they're declared outside the `KWeb(..) { ... }` block then they will be global, inside the block means they will be local to the user's browser session.

Perhaps if you coudl give more details on what you're trying to do I could give more detailed advice.

**Reply by geomagilles** [reply]
Thank you for getting back to me. I want to use Kweb with Ktor with sessions enabled (your [example](https://github.com/kwebio/kweb-demos/blob/master/ktorFeature/src/FeatureApp.kt) does not show that). It's already clearer now, thanks to your comment above. Even if it sounds obvious to you, I believe it's worth a section into the documentation or the FAQ.
